vendingMachine.js

```diff
@@ -1,68 +1,90 @@
-  let result = '';
-  let coord, stoc;
+  const space = " ";
+  const supline = "-";
+  const topLine = supline.repeat(122);
+  const spaceTable = 40; //40 size of spaces for word container in table
+  let itemTable = '';
+  let coord, stock, spaceLeft1, spaceLeft2, spaceRight1, spaceRight2;
+  console.log(topLine);
   for (let i = 0; i < arr1.length; i++) { //0
     for (let j = 0; j < arr1[i].length; j++) { //2
-      result = result + arr1[i][j] + '    ';
+      let a = 0;
+      let left, rigth, squareItem;
+      left = space.repeat(Math.round((spaceTable - (arr1[i][j].length)) / 2));
+      //** residual operator (%)*/
+      //** Returns the integer value resulting from the division of the operands, known as the remainder. */
+      //** In this case it is used to know the length of the array content in that coordinate, and to know if it is odd or even. */
+      if (arr1[i][j].length % 2 === 1) {
+        a = 1;
+      }
+      rigth = space.repeat(Math.round((spaceTable - (arr1[i][j].length)) / 2) - a);
+      squareItem = `|${left}${arr1[i][j]}${rigth}`;
+      if (arr2[i][j] <= 1) {
+        arr1[i][j] = `${supline.repeat(4)}`;
+      }
+      itemTable = itemTable + `${squareItem}`;
       if (j === arr1.length - 1) {
-        coord = ('   ' + (i + 1) + '.' + (j - 1) + '       ' + (i + 1) + '.' + (j) + '       ' + (i + 1) + '.' + (j + 1))
-          //coord = ((i + 1) + '.' + (j - 1) + (i + 1) + '.' + (j) + (i + 1) + '.' + (j + 1))
-        stoc = ('    ' + arr2[i][j - 2] + '         ' + arr2[i][j - 1] + '         ' + arr2[i][j]);
-        //stoc = (arr2[i][j] + arr2[i][j + 1] + arr2[i][j + 2]);
-        result = result + `\n${coord}\n${stoc}\n`;
+        spaceLeft1 = space.repeat(19);
+        spaceRight1 = space.repeat(18);
+        spaceLeft2 = space.repeat(17);
+        spaceRight2 = space.repeat(16);
+        coord = `|${spaceLeft1}${(i + 1)}.${(j - 1)}${spaceRight1}|${spaceLeft1}${(i + 1)}.${(j )}${spaceRight1}|${spaceLeft1}${(i + 1)}.${(j + 1)}${spaceRight1}|`;
+        stock = `|${spaceLeft2}stock:${arr2[i][j - 2]}${spaceRight2}|${spaceLeft2}stock:${arr2[i][j - 1]}${spaceRight2}|${spaceLeft2}stock:${arr2[i][j]}${spaceRight2}|`;
+        itemTable = itemTable + `|\n${stock}\n${coord}\n|${topLine}\n`;
       }
     }
   }
-  console.log(result);
+  console.log(itemTable);
 }
 
-function productStock(arr3, a, b) {
-  if (a <= arr3.length & b <= arr3.length) {
-    let newStock = arr3[a][b];
+function showStock(arr2, row, column) {
+  if (row <= arr2.length & column <= arr2.length) {
+    let newStock = arr2[row][column];
     if (newStock !== 0) {
       newStock = newStock - 1;
-      arr3[a][b] = newStock;
+      arr2[row][column] = newStock;
     } else {
-      console.log('El producto ya no se encutra disponible');
+      console.log('The product is no longer available');
     }
   }
 }
 
-function productRequest(matrix, arr3) {
-  let a, b, code;
+function requestProduct(arr1, arr2) {

```

Comment:
Arr1 and Arr2 what are they?

Reply:
Arr1 and Arr2 referred to the product and stock matrices, the code has been corrected and a more descriptive name has been assigned.